### PR TITLE
fix(goal_planner): fix parking_path curvature and DecidingState transition

### PR DESCRIPTION
**Pull request by soblin**
## Description

- due to a mistake, the curvature of parking_path was calculated from full_path, which caused massive "[WARN 1727950663.638648154] [goal_planner_util]: path.points.size() != curvatures.size() in checkObjectsCollision(). judge as non collision" error
- the time DecidingState started was not properly updated
- transition from NOT_DECIDED to DECIDING was inactivated by mistake

## Related links

**Parent Issue:**

- 

## How was this PR tested?

- https://evaluation.tier4.jp/evaluation/reports/58a47f88-1f99-50dd-acdf-910a2227a75f?project_id=prd_jt
- https://evaluation.tier4.jp/evaluation/reports/51a8cf14-f91a-5b30-876f-663435e9a095?project_id=prd_jt

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/pull_over_planner/pull_over_planner_base.cpp [hunk 1]
```diff
@@ -73,7 +73,7 @@ std::optional<PullOverPath> PullOverPath::create(
   double parking_path_max_curvature{};
   std::tie(full_path_curvatures, full_path_max_curvature) = calculateCurvaturesAndMax(full_path);
   std::tie(parking_path_curvatures, parking_path_max_curvature) =
-    calculateCurvaturesAndMax(full_path);
+    calculateCurvaturesAndMax(parking_path);
```
Comment (soblin) [inline planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/pull_over_planner/pull_over_planner_base.cpp:77]: I was using `full_path` for calculating the curvature of `parking_path` by mistake

File: planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/include/autoware/behavior_path_goal_planner_module/decision_state.hpp [hunk 2]
```diff
@@ -62,11 +62,10 @@ class PathDecisionStateController
...
-  PathDecisionState get_prev_state() const { return prev_state_; }
-
 private:
+  rclcpp::Logger logger_;
+
   PathDecisionState current_state_{};
-  PathDecisionState prev_state_{};
 
   /**
    * @brief update current state and save old current state to prev state
```
Comment (soblin) [inline planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/include/autoware/behavior_path_goal_planner_module/decision_state.hpp:71]: Only `prev_state_.stamp` was used for saving the time when DECIDING state started.
So I deleted `prev_state_` and save `deciding_start_time` instead.

File: planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_planner_module.cpp [hunk 3]
```diff
@@ -2766,10 +2779,15 @@ PathDecisionState PathDecisionStateController::get_next_state(
   // if object recognition for path collision check is enabled, transition to DECIDING to check
   // collision for a certain period of time. Otherwise, transition to DECIDED directly.
   if (parameters.use_object_recognition) {
-    next_state.state = PathDecisionState::DecisionKind::DECIDED;
```
Comment (soblin) [inline planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_planner_module.cpp:2769]: In prior PR, I mistakenly changed the next state to DECIDED. DECIDING is correct for this case.

File: planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_planner_module.cpp [hunk 4]
```diff
@@ -2766,10 +2779,15 @@ PathDecisionState PathDecisionStateController::get_next_state(
...
+    RCLCPP_DEBUG(
+      logger_,
+      "[DecidingPathStatus]: NOT_DECIDED->DECIDING. start checking collision for certain "
+      "period of time");
+    next_state.state = PathDecisionState::DecisionKind::DECIDING;
+    next_state.deciding_start_time = now;
```
Comment (soblin) [inline planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_planner_module.cpp:2787]: fixedto return DECIDING as the next state.

**Review by kosuke55 — APPROVED** [review]

**Pull request merged**